Base/QTApp/qSlicerMainWindow.cxx

```diff
@@ -1299,35 +1312,74 @@ void qSlicerMainWindow::onWarningsOrErrorsOccurred(ctkErrorLogLevel::LogLevel lo
 {
-  ctkSettingsDialog* const settingsDialog =
-    qSlicerApplication::application()->settingsDialog();
-
-  // Reload settings to apply any changes that have been made outside of the
-  // dialog (e.g. changes to module paths due to installing extensions). See
-  // https://github.com/Slicer/Slicer/issues/3658.
-  settingsDialog->reloadSettings();
-
-  // Now show the dialog
-  settingsDialog->exec();
+  qSlicerApplication::application()->openSettingsDialog();
 }
 
 //---------------------------------------------------------------------------
-void qSlicerMainWindow::onNewFileLoaded(const qSlicerIO::IOProperties& fileProperties)
+void qSlicerMainWindow::addFileToRecentFiles(const qSlicerIO::IOProperties& fileProperties)
 {
   Q_D(qSlicerMainWindow);
 
-  d->RecentlyLoadedFileProperties.removeAll(fileProperties);
+  // Remove previous instance of the same file name. Since the file name can be slightly different
+  // (different directory separator, etc.) we don't do a binary compare but compare QFileInfo.
+  QString fileName = fileProperties.value("fileName").toString();
+  if (fileName.isEmpty())
+    {
+    return;
+    }
+  QFileInfo newFileInfo(fileName);
+  for (auto propertiesIt = d->RecentlyLoadedFileProperties.begin(); propertiesIt != d->RecentlyLoadedFileProperties.end() ;)
+    {
+    QFileInfo existingFileInfo(propertiesIt->value("fileName").toString());
+    if (newFileInfo == existingFileInfo)
+      {
+      // remove previous instance
+      propertiesIt = d->RecentlyLoadedFileProperties.erase(propertiesIt);
+      }
+    else
+      {
+      propertiesIt++;
+      }
+    }
 
   d->RecentlyLoadedFileProperties.enqueue(fileProperties);
-
   d->filterRecentlyLoadedFileProperties();
-
   d->setupRecentlyLoadedMenu(d->RecentlyLoadedFileProperties);
-
   // Keep the settings up-to-date
   qSlicerMainWindowPrivate::writeRecentlyLoadedFiles(d->RecentlyLoadedFileProperties);
 }
 
+//---------------------------------------------------------------------------
+void qSlicerMainWindow::onNewFileLoaded(const qSlicerIO::IOProperties& fileProperties)
+{
+  this->addFileToRecentFiles(fileProperties);
+}
+
+//---------------------------------------------------------------------------
+void qSlicerMainWindow::onFileSaved(const qSlicerIO::IOProperties& fileProperties)
+{
+  Q_D(qSlicerMainWindow);
+  QString fileName = fileProperties["fileName"].toString();
+  if (fileName.isEmpty())
+    {
+    return;
+    }
+  QString fileExtension = fileName.toUpper();
+  // Adding every saved file to the recent files list could quickly overwrite the entire list,
+  // therefore we only add the scene file.
+  if (fileName.endsWith(".mrml", Qt::CaseInsensitive)
+    || fileName.endsWith(".mrb", Qt::CaseInsensitive))
+    {
+    // Scene file properties do not contain fileType and it contains screenshot,
+    // which can cause complication when attempted to be stored,
+    // therefore we create a new clean property set.
```

Comment:
```suggestion
    // Since the Scene file properties is not associated with the `fileType` key
    // and can be associated with the `screenshot` key (which can cause complication 
    // when attempted to be stored), we create a new clean property set.

Reply:
By addressing my later comment, we should be able to further simplify and avoid setting `fileType` here.